tests/unit_tests/dft/include/compute_out_of_place.hpp

```diff
@@ -22,32 +22,48 @@
 
 #include "compute_tester.hpp"
 
+template <oneapi::mkl::dft::domain domain>
+std::int64_t get_row_size(const std::vector<std::int64_t> &sizes) noexcept {
+    if constexpr (domain == oneapi::mkl::dft::domain::REAL) {
+        return sizes.back() / 2 + 1;
+    }
+    else {
+        return sizes.back();
+    }
+}
+
 /* Note: There is no implementation for Domain Real */
 template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
 int DFT_Test<precision, domain>::test_out_of_place_buffer() {
     if (!init(MemoryAccessModel::buffer)) {
         return test_skipped;
     }
 
-    const size_t bwd_size = domain == oneapi::mkl::dft::domain::REAL ? (size / 2) + 1 : size;
+    const auto backward_distance = forward_elements;
```

Comment:
```suggestion
    const auto backward_distance = std::accumulate(sizes.begin(), sizes.end() - 1, 1, std::multiplies<>{})*get_row_size(const std::vector<std::int64_t> &sizes);
```
Not sure that the current implementation is what is intended for real transforms otherwise. Multi-dimensional real transforms on GPU would require that for closed-source MKL.
[same comment(s) for USM tests]

Reply:
I'm seeing some weird behaviour when doing this. Specifically If I make the changes show [here](https://github.com/FMarno/oneMKL/compare/multi_d_many_batch_pr...FMarno:oneMKL:multi_d_investigate) then the buffer tests pass but cause the usm tests to fail, although the usm tests are unchanged. Interestingly, if you run the USM tests alone then they still pass.
I'll have a look into this so might take some time to get to your other comments.

Reply:
What kind of failure is it for USM tests when they fail? (exception thrown, crash, incorrect results). Is there a reason we'd want to test different problem configurations between USM and buffers as the changes seem to do? Why not implement the changes for either case?

Reply:
Incorrect results. Here are my test logs 
[real_single_oop.txt](https://github.com/oneapi-src/oneMKL/files/10999498/real_single_oop.txt)
[real_single_oop_USM.txt](https://github.com/oneapi-src/oneMKL/files/10999499/real_single_oop_USM.txt), it is only the 2D single batch tests that are failing. Hopefully its just a problem with the test.
I don't think we want to test different configurations but it seems best to have a quick look into this since its unexpected behaviour.

Reply:
I've emailed you with a minimal reproducer for the problem.
The suggested change is added here https://github.com/oneapi-src/oneMKL/pull/275/commits/14bfce1c072f440e22efaf30a7ee9db5252073f5